rust/parquet/src/arrow/array_reader.rs

```diff
@@ -1000,6 +998,33 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_primitive_array_reader_empty_pages() {
```

Comment:
This is a test for the functionality added in https://github.com/apache/arrow/commit/8a61570a8bd64ae6f6e47ce8efd2287c3c2feded / https://github.com/apache/arrow/pull/7140